adding 2020-10-23-kubernetes-webhooks-made-easy-with-openfaas post

developer-guy opened the pull request:
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

## Description
This PR includes a new blog post about how can we use Kubernetes Validating Admission Webhooks and OpenFaaS together.

## Types of changes

- [x] New blog post
- [ ] Updating an existing blog post
- [ ] Updating part of the page page
- [ ] Adding a new web-page

## Checklist:

- [x] I have given attribution for any images I have used and have permission to use them under Copyright law
- [x] My code follows the [writing-style of the publication](README.md) and I have checked this
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

alexellis commented:
This is very cool.

I noticed that the code example has the same function hard-coded for every event. Can you explain how users can change that, and maybe show two separate functions that do different things, both of which get evaluated for each event?

alexellis commented:
Please have a read of ["Editorial / style guidelines"](https://github.com/openfaas/openfaas.github.io#editorial--style-guidelines) before we review, and check you've hit all the points like image size etc.

Looking forward to trying this myself. 

Also - how would people use this on ARM? If they can't yet, you should say that somewhere, as people will try.

developer-guy commented:
> This is very cool.
> 
> I noticed that the code example has the same function hard-coded for every event. Can you explain how users can change that, and maybe show two separate functions that do different things, both of which get evaluated for each event?

We updated the blog post according to this review, we added a new section for the cloning project and also about how can they extend the project. see: e1ac2f5

alexellis commented on _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:
```diff
@@ -0,0 +1,265 @@
+
+So, in order to apply those requirements to the Kubernetes cluster to ensure the best practices, we can use Kubernetes [ValidatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) and [OpenFaaS](https://www.openfaas.com) together. Since ValidatingAdmissionWebhooks intercepts requests to the apiserver, OpenFaaS functions in ...
+
+Webhook Admission Server is just plain http server that adhere to Kubernetes API. For each Pod create request to the apiserver(I said Pod because we specify which kind of resources that we consider while registering our webhook to the apiserver using [ValidatingWebhookConfiguration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#validatingwebhookconfiguration-v1-admissionreg ...
```
adheres

developer-guy replied:
what should I do to them?

alexellis commented on _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:
```diff
@@ -0,0 +1,265 @@
+<img height="128" src="/images/openfaas/kubernetes.png">
+</p>
+
+## Introduction to Kubernetes Admission webhooks
```
In this section please give a short 1-liner on why you used openfaas for this over doing it with the microservice on its own.

developer-guy replied:
done.

alexellis commented on _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:
```diff
@@ -0,0 +1,265 @@
+title: "Kubernetes Webhooks made easy with OpenFaaS"
+description: "Extending the Kubernetes with Admission webhook using OpenFaaS Serverless Functions"
+date: 2020-10-23
+image: /images/2020-10-github-actions/lego.jpg
```
Please pick your own background for the post and resize as per the README - it gives dimensions

Unplash and pexels are both good websites for royalty free stock photography.

developer-guy replied:
done.

alexellis commented on _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:
```diff
@@ -0,0 +1,265 @@
+  - kubectl
+  - faas-cli
+  - kind
+  - k8s
```
Let's drop kind + k8s.. too many tags.

developer-guy replied:
done.

alexellis commented on _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:
```diff
@@ -0,0 +1,265 @@
+---
+title: "Kubernetes Webhooks made easy with OpenFaaS"
+description: "Extending the Kubernetes with Admission webhook using OpenFaaS Serverless Functions"
```
`In this post you'll learn how to write Kubernetes Admission webhooks using OpenFaaS functions`

developer-guy replied:
done.

alexellis commented on _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:
```diff
@@ -0,0 +1,265 @@
+---
+title: "Kubernetes Webhooks made easy with OpenFaaS"
+description: "Extending the Kubernetes with Admission webhook using OpenFaaS Serverless Functions"
+date: 2020-10-23
```
Put the date to 27th October please and rename the file with `git mv` so the newer date.

developer-guy replied:
done.

alexellis commented on _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:
```diff
@@ -0,0 +1,265 @@
+
+Webhook Admission Server is just plain http server that adhere to Kubernetes API. For each Pod create request to the apiserver(I said Pod because we specify which kind of resources that we consider while registering our webhook to the apiserver using [ValidatingWebhookConfiguration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#validatingwebhookconfiguration-v1-admissionreg ...
+
+* Kubernetes API -> Webhook w/TLS -> OpenFaaS Gateway (w HTTP)
```
One more `->` to the Function

developer-guy replied:
done.

alexellis reviewed (changes requested):
Please see my comments.
